test/format/ops.c

```diff
@@ -481,6 +452,50 @@ snap_check(WT_CURSOR *cursor,
 	return (0);
 }
 
+/*
+ * begin_transaction --
+ *	Choose an isolation configuration and begin a transaction.
+ */
+static u_int
+begin_transaction(TINFO *tinfo, WT_SESSION *session)
+{
+	u_int v;
+	char *config, config_buf[64];
+
+	if ((v = g.c_isolation_flag) == ISOLATION_RANDOM)
+		v = mmrand(&tinfo->rnd, 2, 4);
+	switch (v) {
+	case ISOLATION_READ_UNCOMMITTED:
+		config = "isolation=read-uncommitted";
+		break;
+	case ISOLATION_READ_COMMITTED:
+		config = "isolation=read-committed";
+		break;
+	case ISOLATION_SNAPSHOT:
+	default:
+		v = ISOLATION_SNAPSHOT;
+		config = "isolation=snapshot";
+		if (g.c_txn_timestamps) {
+			/*
+			 * Update the thread's read timestamp with the current
+			 * value to prevent the oldest timestamp moving past our
+			 * allocated timestamp before the commit completes.
+			 */
+			tinfo->read_timestamp = g.timestamp;
```

Comment:
Remove this line since we overwrite it on the next line.

Reply:
That's correct, I think, notice there's an atomic operation between the two assignments. Basically, we set the thread's timestamp (so the timestamp thread won't go past the "current" value), then update the current value (which is an atomic operation so we're guaranteed the first assignment completes before the current value changes), then we assign the result of the atomic operation because that's the timestamp we're really going to use for our reads.

Reply:
I think the comment needs something.  Maybe:

```
Set the thread's read timestamp to the current value before allocating a new read timestamp.
This guarantees that the oldest timestamp won't move past the allocated timestamp before
the transaction begins.
```